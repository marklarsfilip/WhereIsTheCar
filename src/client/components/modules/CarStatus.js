import { useEffect, useState } from 'react';

const CarStatus = ({ status }) => {

    const [oldStatus, setOldStatus] = useState('new');

    useEffect(() => {
        const now = new Date().getTime();
        if (status?.dateTime && now - status.dateTime > 86400000) {
            if (now - status.dateTime > 172800000) {
                setOldStatus('very-old');
            } else {
                setOldStatus('old');
            }
        }
    }, [status])


    return (

        <div className="basis-3/4">
            <h1 className="text-white text-xl mb-5">Bilen är just nu {' '}
                <span className={status.available ? 'text-[#8ee53f]' : 'text-[#d41730]'}>{status.available ? 'ledig' : 'upptagen'}</span>
                <span>{!status.available ? ` av ${status.user}` : ''}</span>
            </h1>
            {!status.available && (
                <>
                    <p className="text-white font-bold mb-3">Beräknad återkomst <span className="block font-normal">{status?.estimatedReturn}</span></p>
                    <p className="text-white font-bold mb-3">Kommentar: <span className="block font-normal">{status?.comment?.length > 0 ? (status.comment) : ('-')}</span></p>
                    {oldStatus === 'old' && (<span className="text-white mt-4">OBS! Denna info registrerades för mer än 24h sedan..</span>)}
                    {oldStatus === 'very-old' && (<span className="text-white mt-4">OBS! Denna info registrerades för flera dagar sedan..</span>)}
                </>
            )}
        </div>


    )
}

export default CarStatus;