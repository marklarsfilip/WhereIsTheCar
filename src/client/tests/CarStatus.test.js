import renderer from 'react-test-renderer';
import CarStatus from '../components/modules/CarStatus';

const mockItem = { available: false, dateTime: 1663153885861 }

it('displays the passed data as status-card', () => {
    const component = renderer.create(
        <CarStatus status={mockItem} />,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});