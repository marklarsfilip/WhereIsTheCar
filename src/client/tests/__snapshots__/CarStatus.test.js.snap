// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays the passed data as status-card 1`] = `
<div
  className="basis-3/4"
>
  <h1
    className="text-white text-xl mb-5"
  >
    Bilen är just nu 
     
    <span
      className="text-[#d41730]"
    >
      upptagen
    </span>
    <span>
       av undefined
    </span>
  </h1>
  <p
    className="text-white font-bold mb-3"
  >
    Beräknad återkomst 
    <span
      className="block font-normal"
    />
  </p>
  <p
    className="text-white font-bold mb-3"
  >
    Kommentar: 
    <span
      className="block font-normal"
    >
      -
    </span>
  </p>
</div>
`;
